tests/components/husqvarna_automower_ble/test_init.py

```diff
@@ -0,0 +1,392 @@
+"""Test the Husqvarna Automower Bluetooth setup."""
+
+import contextlib
+from unittest.mock import Mock
+
+from bleak import BleakError
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.husqvarna_automower_ble.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+from . import AUTOMOWER_SERVICE_INFO
+
+from tests.common import MockConfigEntry
+
+pytestmark = pytest.mark.usefixtures("mock_client")
+
+
+async def test_setup(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    mock_entry: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test setup creates expected devices."""
+
+    mock_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert mock_entry.state is ConfigEntryState.LOADED
+
+    device = device_registry.async_get_device(
+        identifiers={(DOMAIN, AUTOMOWER_SERVICE_INFO.address)}
+    )
+    assert device == snapshot
+
+    coordinator = mock_entry.runtime_data.coordinator
+    await coordinator.async_shutdown()
```

Comment:
Don't touch internals. Rather check if we really did the things needed for a shutdown at the (soon to be added) mocks